lib/logstash/filters/cidr.rb

```diff
@@ -35,11 +35,82 @@ class LogStash::Filters::CIDR < LogStash::Filters::Base
   #     }
   config :network, :validate => :array, :default => []
 
+  # The full path of the external file containing the IP network(s) to check against. Example:
+  # [source,ruby]
+  #     filter {
+  #       %PLUGIN% {
+  #         add_tag => [ "linklocal" ]
+  #         address => [ "%{clientip}" ]
+  #         network_path => "/etc/logstash/networks"
+  #       }
+  #     }
+  # NOTE: it is an error to specify both 'network' and 'network_path'.
+  config :network_path, :validate => :path
+
+  # The separator character used in the encoding of the external file pointed by network_path.
+  config :separator,:validate => :string, :default => "\n"
+
+  # When using a network list from a file, this setting will indicate how frequently
+  # (in seconds) logstash will check the file for updates.
+  config :refresh_interval, :validate => :number, :default => 600
+
   public
-  def register
-    # Nothing
+  def register    
+    rw_lock = java.util.concurrent.locks.ReentrantReadWriteLock.new
+    @read_lock = rw_lock.readLock
+
+    if @network_path && !@network.empty?
+    	raise LogStash::ConfigurationError, I18n.t(
+    	"logstash.agent.configuration.invalid_plugin_register",
+    	:plugin => "filter",
+    	:type => "cidr",
+    	:error => "The configuration options 'network' and 'network_path' are mutually exclusive"
+    	)
+    end
+ 
+    if @network_path
+      @next_refresh = Time.now + @refresh_interval
+      lock_for_read { load_file }
+    end
   end # def register
 
+  def lock_for_read #ensuring only one thread updates the network block list
+    @read_lock.lock
+    begin
+      yield
+    ensure
+      @read_lock.unlock
+    end
+  end #define lock_for_read
+
+  def needs_refresh?
+    @next_refresh < Time.now
+  end # define needs_refresh
+
+  def load_file
+    if @network && !File.exists?(@network_path)
+      @logger.warn("file read error, continuing with old version")
```

Comment:
My reasoning was: If a network list was already in place (because of a past successful parsing) and the file was now unreadable. Then the filter would just stay with the old version.
If the filter was never able to parse the networks from the file, then @network would be null, resulting in an error in later execution of the filter.
What would be the best way to handle this? This is the first time I've contributed to FOSS and also my first time with Ruby